packages/shared_preferences/shared_preferences_macos/example/test_driver/shared_preferences_e2e.dart

```diff
@@ -0,0 +1,89 @@
+import 'dart:async';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:shared_preferences/shared_preferences.dart';
+import 'package:e2e/e2e.dart';
+
+void main() {
+  E2EWidgetsFlutterBinding.ensureInitialized();
```

Comment:
I think that e2e tests should live in the app-facing plugin, not the platform implementation. Otherwise they'll be duplicated for every platform.

Reply:
But these are the only test we have for the package. IIUC, If somebody edits the package and breaks it, the app-facing plugin wouldn't detect it, is that right? 

If we don't need them here, then we don't need the example app at all since that was it's only purpose.

Reply:
I think it makes sense to have the app-facing plugin's example app include every endorsed platform, not just iOS and Android.

For plugins that have lightweight e2e integration tests like this one, we can teach our CI to run the them using the app-facing plugin's example app for every endorsed platform implementation, using a path dependency. That doesn't need to happen to land this PR, but I think we should file an issue and start heading in that direction.

We can also have heavyweight integration tests involving a native UI / lifecycle interaction, and in fact I'm planning to land an Espresso test of shared_preferences as soon as https://github.com/flutter/plugins/pull/2369 lands. I think this test should still live in the app-facing plugin though, so it can share the Dart code of the example app.

@amirh may have opinions about this as well

Reply:
Yeah, this pr adds macos testing on the app-facing plugin. https://github.com/flutter/plugins/pull/2386 This example app was suggested by @amirh to be able to test the macos package and be "self contained". I'm wondering if we need them since once that pr is landed, any changes to the macos plugin will trigger the app-facing tests which will run the macos implementation.

Reply:
This is fine for now but let's figure out how to share example app code in the future. It seems like we may be able to do it with dependency overrides.

Filed https://github.com/flutter/flutter/issues/46716 (maybe we can reference that in a TODO?)